File: programmers/고득점 Kit/완전탐색/42840.py
# Solved (20m)
from collections import defaultdict
import math
def solution(answers):
    answer = []
    one = [1, 2, 3, 4, 5] * math.ceil(len(answers)/5)
    two = [2, 1, 2, 3, 2, 4, 2, 5] * math.ceil(len(answers)/8)
    three = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5] * math.ceil(len(answers)/10)
    score = defaultdict(int)
    for i, ans in enumerate(answers):
        if one[i]==ans:
            score[1] += 1
        if two[i]==ans:
            score[2] += 1
        if three[i]==ans:
            score[3] += 1
            
    for student, score in sorted(score.items(), key=lambda x:x[1], reverse=True):
        if not answer or answer[-1][1]==score:
            answer.append((student, score))
        else:
            break        
    return sorted([x[0] for x in answer])


# 다른 사람 풀이
# -> defaultdict말고 list로 고정된 index로 카운팅하면 정렬 필요 없음
def solution(answers):
    answer = []
    pattern1 = [1, 2, 3, 4, 5]
    pattern2 = [2, 1, 2, 3, 2, 4, 2, 5]
    pattern3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    score = [0, 0, 0]
    for i, ans in enumerate(answers):
        if pattern1[i%len(pattern1)]==ans:
            score[0] += 1
        if pattern2[i%len(pattern2)]==ans:
            score[1] += 1
        if pattern3[i%len(pattern3)]==ans:
            score[2] += 1
            
    for idx, s in enumerate(score):
        if s == max(score):
            answer.append(idx+1)
    return answer

File: programmers/고득점 Kit/완전탐색/42840.java
import java.util.*;
class Solution {
    public int[] solution(int[] answers) {
        int[] one = {1,2,3,4,5};
        int[] two = {2,1,2,3,2,4,2,5};
        int[] three = {3,3,1,1,2,2,4,4,5,5};
        int[] score = {0, 0, 0};
        for(int i=0; i<answers.length; i++){
            if(answers[i]==one[i%one.length]) score[0]++;
            if(answers[i]==two[i%two.length]) score[1]++;
            if(answers[i]==three[i%three.length]) score[2]++;
        }

        int max = Math.max(score[0], Math.max(score[1], score[2]));
        ArrayList<Integer> answer = new ArrayList<Integer>();

        for (int i = 0; i < 3; i++) {
            if (score[i] == max) {
                answer.add(i + 1);
            }
        }

        // ArrayList<Integer> → int[] 변환
        return answer.stream().mapToInt(i -> i).toArray();
    }
}